File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.java
import java.util.*;

class Solution {
    public int minMirrorPairDistance(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int ans = Integer.MAX_VALUE;

        for (int i = 0; i < nums.length; i++) {
            int rev = reverseNum(nums[i]);

            // check if current number matches any previous reverse
            if (map.containsKey(nums[i])) {
                ans = Math.min(ans, i - map.get(nums[i]));
            }

            // store reverse instead of original
            map.put(rev, i);
        }

        return ans == Integer.MAX_VALUE ? -1 : ans;
    }

    public int reverseNum(int num) {
        int reverse = 0;
        while (num != 0) {
            int digit = num % 10;
            reverse = reverse * 10 + digit;
            num /= 10;
        }
        return reverse;
    }
}

File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.py
class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        him={}
        ans=float('inf')
        for i in range(len(nums)):
            rev=self.reverseNum(nums[i])
            if nums[i] in him:
                ans=min(abs(i-him[nums[i]]),ans)
            him[rev]=i
        return ans if ans!=float('inf') else -1
    def reverseNum(self, num: int) -> int:
        rev = 0
        while num != 0:
            digit = num % 10
            rev = rev * 10 + digit
            num //= 10
        return rev